packages/eas-cli/src/commands/build/index.ts

```diff
@@ -115,7 +128,51 @@ export default class Build extends EasCommand {
       Errors.error('--json is allowed only when building in non-interactive mode', { exit: 1 });
     }
 
+    const validPlatformToBuildResourceClassMapping = new Map(
```

Comment:
You don't need Map here, it can be just a plain object

Reply:
I'm going to remove this logic anyway, in light of making the flag options non-platform specific. I'll move the resource-class determination into the existing buildProfile generation logic, which should make things a bit cleaner.